File: src/main/scala/com/github/chengpohi/script/EDQLRunResult.scala
package com.github.chengpohi.script

import scala.util.{Success, Try}

case class EDQLRunResult(response: Try[Seq[String]], context: EDQLRunContext = null) {

  def isSuccess: Boolean = response.isSuccess

  def isFail: Boolean = response.isFailure

  def failed: Throwable = response.failed.get

  def success: Seq[String] = response.get
}

object EDQLRunResult {
  def apply(response: Try[Seq[String]]): EDQLRunResult = {
    new EDQLRunResult(response)
  }

  def apply(response: Seq[String], runContext: EDQLRunContext): EDQLRunResult = {
    new EDQLRunResult(Success(response), runContext)
  }
}

File: src/main/java/com/github/chengpohi/script/EDQLRunResult.kt
package com.github.chengpohi.script

data class EDQLRunResult(
    val response: Result<List<String>>,
    val context: EDQLRunContext? = null
) {
    val isSuccess: Boolean
        get() = response.isSuccess

    val isFail: Boolean
        get() = response.isFailure

    val failed: Throwable
        get() = response.exceptionOrNull() ?: throw IllegalStateException("No exception available")

    val success: List<String>
        get() = response.getOrThrow()
}